File: i-payout/recipes/js/payout/hosted.js
const base64 = require('base-64');
const axios = require('axios');

const baseUrl = "https://merchantapi.testewallet.com/api/v1";

const username = "<username>";  // Replace with your actual username
const password = "<password>";  // Replace with your actual password
const authStr = `${username}:${password}`;
const encodedAuthStr = base64.encode(authStr);

const tokenUrl = `${baseUrl}/Authentication/Login`;
let headers = {
    "accept": "application/json",
    "authorization": `Basic ${encodedAuthStr}`,
    "X-MerchantId": username,
};

axios.get(tokenUrl, { headers: headers })
    .then(response => {
        const apiToken = response.data.data.token;

        const beneficiaryUrl = `${baseUrl}/beneficiary/create`;
        const beneficiaryBody = {
            "username": "john_doe",
            "firstName": "John",
            "lastName": "Doe",
            "emailAddress": "johndoe@gmail.com",
        };
        headers = {
            "Authorization": `Bearer ${apiToken}`,
            "X-MerchantId": username,
            "accept": "application/json",
            "content-type": "application/*+json",
        };

        return axios.post(beneficiaryUrl, beneficiaryBody, { headers: headers });
    })
    .then(() => {
        const payoutUrl = `${baseUrl}/PayOut/create`;
        const payoutBody = {
            "partnerBatchId": "batch_001",
            "allowDuplicates": true,
            "autoLoad": true,
            "currencyCode": "USD",
            "arrAccounts": [
                { "username": "john_doe", "amount": 200, "merchantReferenceId": "789012" }
            ],
        };
        headers = {
            "Authorization": `Bearer ${apiToken}`,
            "X-MerchantId": username,
            "Content-Type": "application/json",
        };

        return axios.post(payoutUrl, payoutBody, { headers: headers });
    })
    .then(response => {
        console.log(response.data);
    })
    .catch(error => {
        console.error(error);
    });

File: recipes/python/old/hosted.py
import base64
import requests

base_url = "https://merchantapi.testewallet.com/api/v1"

username = "6d099995-e2f9-4a01-bef1-258eb99c1b77"
password = "Ys=LYgysz0s+tZPC"
auth_str = f"{username}:{password}"
encoded_auth_str = base64.b64encode(auth_str.encode()).decode()

token_url = f"{base_url}/Authentication/Login"
headers = {
    "accept": "application/json",
    "authorization": f"Basic {encoded_auth_str}",
    "X-MerchantId": username,
}

response = requests.get(token_url, headers=headers)
api_token = response.json()["data"]["token"]


beneficiary_url = f"{base_url}/beneficiary/create"
body = {
    "username": "john_doe",
    "firstName": "John",
    "lastName": "Doe",
    "emailAddress": "johndoe@gmail.com",
}
headers = {
    "Authorization": f"Bearer {api_token}",
    "X-MerchantId": username,
    "accept": "application/json",
    "content-type": "application/*+json",
}

response = requests.post(beneficiary_url, headers=headers, json=body)

payout_url = f"{base_url}/PayOut/create"
body = {
    "partnerBatchId": "batch_001",
    "allowDuplicates": True,
    "autoLoad": True,
    "currencyCode": "USD",
    "arrAccounts": [
        {"username": "john_doe", "amount": 200, "merchantReferenceId": "789012"}
    ],
}
headers = {
    "Authorization": f"Bearer {api_token}",
    "X-MerchantId": username,
    "Content-Type": "application/json",
}

response = requests.post(payout_url, headers=headers, json=body)

print(response.json())
